cmd/sdkstats/main.go

```diff
@@ -7,46 +7,170 @@ import (
+		&oauth2.Token{AccessToken: token},
 	)
 	tc := oauth2.NewClient(ctx, ts)
-	client := github.NewClient(tc)
+	client := sdkstats.Client{Client: github.NewClient(tc)}
 
-	var queries []string
-
-	queries = append(queries, "filename:Dockerfile quay.io/operator-framework/helm-operator")
-	queries = append(queries, "filename:Dockerfile quay.io/operator-framework/ansible-operator")
-	queries = append(queries, "filename:go.mod github.com/operator-framework/operator-sdk")
-	queries = append(queries, "filename:Gopkg.toml github.com/operator-framework/operator-sdk")
+	queries := []sdkstats.RepoMetadataQuery{
+		sdkstats.RepoMetadataQuery{
+			ProjectType: "helm",
+			Queries:     []string{"filename:Dockerfile quay.io/operator-framework/helm-operator"},
+			VersionParser: &baseVersionParser{
+				searchQ: "quay.io/operator-framework/helm-operator",
+			},
+		},
+		sdkstats.RepoMetadataQuery{
+			ProjectType: "ansible",
+			Queries:     []string{"filename:Dockerfile quay.io/operator-framework/ansible-operator"},
+			VersionParser: &baseVersionParser{
+				searchQ: "quay.io/operator-framework/ansible-operator",
+			},
+		},
+		/*sdkstats.RepoMetadataQuery{
+			ProjectType: "go.mod",
+			Queries: []string{
+				"filename:go.mod github.com/operator-framework/operator-sdk",
+			},
+			VersionParser: &gomodVersionParser{
+				searchQ: "replace github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk v",
+			},
+		},
+		/*sdkstats.RepoMetadataQuery{
+			ProjectType:   "gopkg.toml",
+			Queries:       []string{"filename:Gopkg.toml github.com/operator-framework/operator-sdk"},
+			VersionParser: &unknownVersionParser{},
+		},*/
+	}
+	// GetStats function for Query String from 'queries', These Strings are specific to Operator-SDK patterns.
+	collectStats := [][]sdkstats.RepoMetadata{}
 
 	for _, r := range queries {
-		stats, err := sdkstats.GetStats(client, r)
-		if len(stats) == 0 {
-			log.Fatal("GITHUB API is down at time, Please try after few minutes")
-		} else {
-			fileName := stats[0].ProjectType + ".json"
-			if err != nil {
-				fmt.Println(err)
+		stats, err := client.GetStats(ctx, r)
+
+		fmt.Println("Total count for ", r.ProjectType, ":", len(stats))
+		if err != nil {
+			fmt.Printf("Failed to get stats for queries %v: %v\n", r.Queries, err)
+		}
+		collectStats = append(collectStats, stats)
+	}
+	fileName := "Search_Results.json"
+	file, err := json.MarshalIndent(collectStats, "", " ")
+	err = ioutil.WriteFile(fileName, file, 0644)
+	if err != nil {
+		fmt.Printf("Error encountered while writing results to file %v\n", err)
+		fmt.Println(collectStats)
+		os.Exit(1)
+	}
+	fmt.Println("Results are written in Search_Results.json")
+}
+
+// Parse the given Code result to search Text Matches for Version number.
+func (p baseVersionParser) ParseVersion(codeResults github.CodeResult) (string, error) {
+	baseImageRegex := regexp.QuoteMeta(p.searchQ)
+	versionRegex := strings.Trim(`(:([^s]+\n))?`, "\n")
```

Comment:
I think you want the `\n` newline character to be appended _after_ the version capture group, basically after the very end after the `?`. Also don't think you need `strings.Trim()` here.